### DISCUSSION TITLE
Any chance of adding a proper AGC to the receiver?

### DISCUSSION THREAD

**Discussion by loughkb** [AlexandreRouma/SDRPlusPlus#366] [Ideas]
Howdy,

    When I'm using SDR++, Especially when doing some shortwave listening,  I find that I really miss having a proper AGC.   The AGC function in GQRX is a good model of helpful function.    It can make listening to shortwave AM stations so much more pleasurable.  

    To my ears, it almost seems like a sort of fast AGC is already going on, but may be way too fast, or driven by audio rather than the RF coming through the IF filter.   I was listening to WWV during a test and noticed that the gain seemed to bring up the noise between the 'thump' sound WWV ticks off the seconds with.   I've also noticed it when receiving a fairly strong CW transmission.

    If you really want to get  fancy and stand out among several of the other SDR programs,  you could also implement some advanced filter functions like a notch function.   Sometimes when I'm listening to Radio New Zealand,  There will be a carrier sitting right in the middle of the audio passband that I wish I could notch out.    

    Anyway,  just a thought.  Take it or leave it, it's still a damn nice program.

Kevin

**Comment by AlexandreRouma** [comment]
hello,

This was already requested a while back  but I haven't gotten around to it yet. The current AGC is a block based AGC that tries to keep the gain maximum all the time, it's not really ideal for audio but my goal was just to have something that sounded decent and worked for digital modes.

As for a notch filter, I haven't planned for it yet. Tight notch filters can be extremely computationally intensive and the UI isn't really adequate to intuitively operate it. Since the DSP of SDR++ is fully from scratch, it's something that'll take a while to get right...
I'll have to think about it in the future

**Comment by loughkb** [comment]
Cool, thanks for the info.   I'll check the git now and then to watch for new versions.
Thanks for the program.  I'm enjoying it.

**Comment by pbij2001** [comment]
I think "just" a speed slider to the decay time would already do huge miracles to the listening experience.
Just between marks as you do all the hard work :-)
Tnx for that!
